Libs/MRML/Core/Testing/vtkMRMLDoubleArrayNodeTest1.cxx

```diff
@@ -10,18 +10,413 @@
 
 =========================================================================auto=*/
 
+// MRML includes
+#include "vtkMRMLCoreTestingMacros.h"
 #include "vtkMRMLDoubleArrayNode.h"
+#include "vtkMRMLDoubleArrayStorageNode.h"
 
+// VTK includes
+#include <vtkNew.h>
 
-#include "vtkMRMLCoreTestingMacros.h"
+// STD includes
+#include <vector>
+#include <sstream>
+
+//---------------------------------------------------------------------------
+bool TestSetValues();
+bool TestSetValue();
+bool TestSetXYValue();
+bool TestSetXYerrValue();
+bool TestAddValues();
+bool TestAddValue();
+bool TestAddXYValue();
+bool TestAddXYerrValue();
+bool TestReadFileWithLabels(std::string filepath);
+bool TestReadFileWithoutLabels(std::string filepath);
+bool TestReadOldFile(std::string filepath);
+
+//---------------------------------------------------------------------------
+int vtkMRMLDoubleArrayNodeTest1(int argc, char * argv[])
+{
+  vtkNew<vtkMRMLDoubleArrayNode> node1;
+
+  EXERCISE_BASIC_OBJECT_METHODS(node1.GetPointer());
+
+  EXERCISE_BASIC_MRML_METHODS(vtkMRMLDoubleArrayNode, node1.GetPointer());
+
+  if (argc != 2)
+    {
+    std::cerr << "Line " << __LINE__
+	      << " - Missing parameters !\n"
+	      << "Usage: " << argv[0] << " /path/to/temp"
+	      << std::endl;
+    return EXIT_FAILURE;
+    }
+
+  const char* tempDir = argv[1];
+
+  std::string fileNameOldFile = std::string(tempDir) + "/vtkMRMLDoubleArrayNodeTest1_OldFile.mcsv";
```

Comment:
They do not exist at the beginning, they are only created by the storage node writer [1][2] or manually to replicate the former writer.
I could maybe create a .mcsv file for testing the backward compatibility in `TestReadOldFile` [3] instead of writing it to test the backward compatibility?

  [1] https://github.com/Slicer/Slicer/pull/328/files#diff-ecf042206b75393c4c26a631735fc225R281
  [2] https://github.com/Slicer/Slicer/pull/328/files#diff-ecf042206b75393c4c26a631735fc225R323
  [3] https://github.com/Slicer/Slicer/pull/328/files#diff-ecf042206b75393c4c26a631735fc225R361